hledger-lib/Hledger/Reports/BudgetReport.hs

```diff
@@ -218,23 +218,23 @@ combineBudgetAndActual ropts j
         totActualByPeriod = Map.fromList $ zip actualperiods actualtots :: Map DateSpan Change
 
 -- | Render a budget report as plain text suitable for console output.
-budgetReportAsText :: ReportOpts -> BudgetReport -> String
-budgetReportAsText ropts@ReportOpts{..} budgetr =
-    title ++ "\n\n" ++
-    renderTable def{tableBorders=False,prettyTable=pretty_tables_}
+budgetReportAsText :: ReportOpts -> BudgetReport -> TL.Text
```

Comment:
The reason here is that Text builders are unpacked to lazy Text, due to how they are constructed. They can then be made strict to get strict Text. Since the options were to either convert to lazy Text and then use the lazy Text putStr, or to convert to lazy Text, convert that to strict Text, and then use the strict Text putStr, I thought it might be reasonable to do a couple fewer steps. That said, it's ugly and cumbersome to have both strict and lazy Text values floating around, so if there's not a big performance hit it might be reasonable to make them strict.

Reply:
See the section on Builders [here](https://www.fpcomplete.com/haskell/tutorial/string-types/).